sql/2021/caching/cache_control_directives.sql

```diff
@@ -0,0 +1,71 @@
+  COUNTIF(uses_pre_check_zero AND uses_post_check_zero) AS total_using_pre_check_zero_and_post_check_zero,
+  COUNTIF(uses_pre_check_zero) AS total_using_pre_check_zero,
+  COUNTIF(uses_post_check_zero) AS total_using_post_check_zero,
+  COUNTIF(uses_cache_control AND NOT uses_max_age AND NOT uses_no_cache AND NOT uses_public AND NOT uses_must_revalidate AND NOT uses_no_store AND NOT uses_private AND NOT uses_proxy_revalidate AND NOT uses_s_maxage AND NOT uses_no_transform AND NOT uses_immutable AND NOT uses_stale_while_revalidate AND NOT uses_stale_if_error AND NOT uses_pre_check_zero AND NOT uses_post_check_zero) AS total_erroneous_directives,
+  COUNTIF(uses_cache_control) / COUNT(0) AS pct_using_cache_control,
+  COUNTIF(uses_max_age) / COUNT(0) AS pct_using_max_age,
+  COUNTIF(uses_no_cache) / COUNT(0) AS pct_using_no_cache,
+  COUNTIF(uses_public) / COUNT(0) AS pct_using_public,
+  COUNTIF(uses_must_revalidate) / COUNT(0) AS pct_using_must_revalidate,
+  COUNTIF(uses_no_store) / COUNT(0) AS pct_using_no_store,
+  COUNTIF(uses_private) / COUNT(0) AS pct_using_private,
+  COUNTIF(uses_proxy_revalidate) / COUNT(0) AS pct_using_proxy_revalidate,
+  COUNTIF(uses_s_maxage) / COUNT(0) AS pct_using_s_maxage,
+  COUNTIF(uses_no_transform) / COUNT(0) AS pct_using_no_transform,
+  COUNTIF(uses_immutable) / COUNT(0) AS pct_using_immutable,
+  COUNTIF(uses_stale_while_revalidate) / COUNT(0) AS pct_using_stale_while_revalidate,
+  COUNTIF(uses_stale_if_error) / COUNT(0) AS pct_using_stale_if_error,
+  COUNTIF(uses_no_store AND uses_no_cache AND uses_max_age_zero) / COUNT(0) AS pct_using_no_store_and_no_cache_and_max_age_zero,
+  COUNTIF(uses_no_store AND uses_no_cache AND NOT uses_max_age_zero) / COUNT(0) AS pct_using_no_store_and_no_cache_only,
+  COUNTIF(uses_no_store AND NOT uses_no_cache AND NOT uses_max_age_zero) / COUNT(0) AS pct_using_no_store_only,
+  COUNTIF(uses_max_age_zero AND NOT uses_no_store) / COUNT(0) AS pct_using_max_age_zero_without_no_store,
+  COUNTIF(uses_pre_check_zero AND uses_post_check_zero) / COUNT(0) AS pct_using_pre_check_zero_and_post_check_zero,
+  COUNTIF(uses_pre_check_zero) / COUNT(0) AS pct_using_pre_check_zero,
+  COUNTIF(uses_post_check_zero) / COUNT(0) AS pct_using_post_check_zero,
+  COUNTIF(uses_cache_control AND NOT uses_max_age AND NOT uses_no_cache AND NOT uses_public AND NOT uses_must_revalidate AND NOT uses_no_store AND NOT uses_private AND NOT uses_proxy_revalidate AND NOT uses_s_maxage AND NOT uses_no_transform AND NOT uses_immutable AND NOT uses_stale_while_revalidate AND NOT uses_stale_if_error AND NOT uses_pre_check_zero AND NOT uses_post_check_zero) / COUNT(0) AS pct_erroneous_directives
+FROM (
+  SELECT
+    _TABLE_SUFFIX AS client,
+    TRIM(resp_cache_control) != "" AS uses_cache_control,
+    REGEXP_CONTAINS(resp_cache_control, r'(?i)max-age\s*=\s*[0-9]+') AS uses_max_age,
+    REGEXP_CONTAINS(resp_cache_control, r'(?i)max-age\s*=\s*0') AS uses_max_age_zero,
```

Comment:
Yes but my point is it's currently NOT being included in `uses_max_age` nor `uses_max_age_zero`. Certainly should be in first and maybe should be in second (since it's effectively the same as 0).

Reply:
Oh I understand now. Still not sure if we want to include invalid usage? For example, should we also count `max-age=foo`? If so we should drop the `\s*[0-9]+` requirement from the regex.

Reply:
Ah I presume negative was valid and the same as zero (in same way as expired can be set to old date). But maybe I'm wrong about that! RFC hunting time...

Reply:
Nope you are correct. It has to be a positive number. Resolving.